### ISSUE TITLE
Implement the special relation `dj.U` 

### ISSUE THREAD

**Issue by dimitri-yatsenko** [datajoint/datajoint-python#112]
`dj.U(*attributes)` represents the relation comprising all possible values of its attributes.  It's initialized with the list of its attributes.  Its primary key comprises all its attributes.  `dj.U()` cannot be fetched from directly and is not represented in the database but it is useful in may relational expressions.

Let the relation `trials` represent some stimulus trials and `'contrast'` and `'brightness'` are two of its attributes.  Then 
```python
dj.U('contrast') & trials   # contains all unique values of contrast found in trials.
dj.U('brightness', 'contrast') & trials   # contains all unique combinations of contrast/brightness found in trials.
dj.U('contrast').aggregate(trials, 'count(*)->n')   # contains the number of trials for each unique contrast
```

Prior to `dj.U`, the same functionality could be achieved by creating a dedicated lookup table containing all values of contrast (for the first example) or combinations of brightness and contrast (for the second example).  `dj.U` allows avoiding creating extra tables that are required only for the convenience of relational expressions.

Here is a more complex example.  If there are two relations containing trials: `trials1` and `trials2`.  
```python
(dj.U('contrast') & trials2) - trials1  # contains all contrasts from trials2 that are not found in trials1
```

The special case with the empty attribute list `dj.U()`  represents the relation with one tuple that has no attributes.  As such, it matches any tuple.  It can be thought as the identity relation with respect to join and restrict:  `rel*dj.U() == rel` and `rel & dj.U() == rel`.  It becomes useful for the aggregation of an entire relation:
```python
dj.U().aggregate(trials, 'count(distinct contrast)')    # count the number of distinct contrasts in trials
```

**Comment by dimitri-yatsenko** [comment]
I am thinking to call this relation `dj.U` (for "universal") rather than `dj.All`.

**Comment by eywalker** [comment]
@dimitri-yatsenko has already convinced me that `dj.U` is a good choice.

**Comment by fabiansinz** [comment]
Agreed. it's a good name

**Comment by dimitri-yatsenko** [comment]
implemented in 4debbd88f
